declarative-config-bridge/src/main/java/io/opentelemetry/instrumentation/config/bridge/ConfigPropertiesBackedDeclarativeConfigProperties.java

```diff
@@ -0,0 +1,207 @@
+   *
+   * <p>This will be misleading if anyone is comparing the return value to null.
+   */
+  @Override
+  public DeclarativeConfigProperties getStructured(String name) {
+    List<String> newPath = new ArrayList<>(path);
+    newPath.add(name);
+    return new ConfigPropertiesBackedDeclarativeConfigProperties(configProperties, newPath);
+  }
+
+  @Nullable
+  @Override
+  @SuppressWarnings("unchecked") // Safe because T is known to be String via scalarType check
+  public <T> List<T> getScalarList(String name, Class<T> scalarType) {
+    if (scalarType != String.class) {
+      return null;
+    }
+    String fullPath = pathWithName(name);
+
+    // Check explicit list mappings first
+    String mappedKey = LIST_MAPPINGS.get(fullPath);
+    if (mappedKey != null) {
+      List<String> list = configProperties.getList(mappedKey);
+      if (!list.isEmpty()) {
+        return (List<T>) list;
+      }
+      return null;
+    }
+
+    // Standard mapping
+    List<String> list = configProperties.getList(toPropertyKey(fullPath));
+    if (list.isEmpty()) {
+      return null;
+    }
+    return (List<T>) list;
+  }
+
+  @Nullable
+  @Override
+  public List<DeclarativeConfigProperties> getStructuredList(String name) {
+    String fullPath = pathWithName(name);
+    if (GENERAL_PEER_SERVICE_MAPPING.equals(fullPath)) {
+      return PeerServiceMapping.getList(configProperties);
+    }
+    return null;
+  }
+
+  @Override
+  public Set<String> getPropertyKeys() {
+    // this is not supported when using system properties based configuration
+    return emptySet();
+  }
+
+  @Override
+  public ComponentLoader getComponentLoader() {
+    return configProperties.getComponentLoader();
+  }
+
+  private String pathWithName(String name) {
+    if (path.isEmpty()) {
+      return name;
+    }
+    return String.join(".", path) + "." + name;
+  }
+
+  private static String toPropertyKey(String fullPath) {
+    String translatedPath = translatePath(fullPath);
+
+    // Handle agent prefix: java.agent.* → otel.javaagent.*
+    if (translatedPath.startsWith("agent.")) {
+      return "otel.java" + translatedPath;
+    }
+
+    // Handle jmx prefix: java.jmx.* → otel.jmx.*
+    if (translatedPath.startsWith("jmx.")) {
+      return "otel." + translatedPath;
+    }
+
+    // Standard mapping
+    return "otel.instrumentation." + translatedPath;
+  }
+
+  private static String translatePath(String path) {
+    StringBuilder result = new StringBuilder();
+    for (String segment : path.split("\\.")) {
+      // Skip "java" segment - it doesn't exist in system properties
+      if ("java".equals(segment)) {
+        continue;
+      }
+      if (result.length() > 0) {
+        result.append(".");
+      }
+      result.append(translateName(segment));
+    }
+    return result.toString();
+  }
+
+  private static String translateName(String name) {
+    if (name.endsWith("/development")) {
+      return "experimental."
```

Comment:
there's also the case of `experimental` in the middle - which I addressed by https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/15622/changes#diff-e9355f50185447f38cce7dc72f64f13f37adcac531399b357e3f507b619d8c3cR152

It's used in `*.emit-experimental-telemetry` only, which I propose to be translated to `emit-experimental-telemetry/development.